client/my-sites/stats/activity-log/index.jsx

```diff
@@ -47,10 +56,37 @@ class ActivityLog extends Component {
 		translate: PropTypes.func.isRequired,
 	};
 
+	state = {
+		requestedRestoreTimestamp: null,
+		showRestoreConfirmDialog: false,
```

Comment:
That's what I thought too 😀 

The problem is that the dialog has a transition, so if you don't separate visibility from data in the dialog, there are `200ms` when closing the dialog where the timestamp is lost and `Invalid date` is displayed. This seemed the most straightforward solution.